src/vector.lua

```diff
@@ -0,0 +1,37 @@
+local vector = {}
+
+local function sign(x)
```

Comment:
I put sign in with the vector functions. It actually doesn't have anything to do with vectors inherently, but it's almost always used in code that deals with vectors. That's a poor argument for putting it here, I know...